### DISCUSSION TITLE
Disable private text messaging to prevent spam

### DISCUSSION THREAD

**Discussion by bear101** [BearWare/TeamTalk5#1806] [Ideas]
It seems the preferred way to spam TeamTalk servers is to send private text messages to all users. Should the ability to send private text messages be a user-right in "User Accounts" dialog? Or are people ok with the option on client-side to disable private text messaging?

**Comment by CoBC** [comment]
Probably good to have an user-right for this. Some users maybe don't know how to disable text messaging in there client, and on a server with a lot of users it can be ennoying if all users have to disable messages for someone.

**Comment by Elshara** [comment]
I think it should be a hybrid of the two. Just prompt people by default if they want private messages to be delivered to them from users of the connected server. And let them know you could add exceptions to be either blacklisted or whitelisted for blocked or allowed users respectively if necessary under settings.
Here's a scenario. I am user A. I login to Team Talk on Server 1. User B, C and D are online in server 1. User B, I don't know. User C, I do. And user D is friends with user C.
Assume I as user A hang out quite frequently on Server 1. But not all users connected to Server 1 know this. User D is such a user. But I don't know user D yet but Users B and C do.
A block on all private messages means that User D can't get to know the only person they may not know on a particular friend's server, using Server 1 as an example. Allowing all private messages means that I can't block user D from getting to know me as I'm user A. Furthermore, Users B and C may get upset if I'm uninterested in what user D has to say, therefore causing needless head aches. Because blocking private messages will just mean the public chat in Server 1 would get used instead to draw attention. Yes you can disable messaging altogether in preferences, but introducing a block on all private messages by default will defeat the purpose of Team Talk in the first place. It is not a silent hang out zone. There needs to be mutual intelligibility.

**Comment by beqabeqa473** [comment]
In case of user rights, i think there should be a possibility to
configure also channel messages

On 4/7/23, Elshara Silverheart ***@***.***> wrote:
> I think it should be a hybrid of the two. Just prompt people by default if
> they want private messages to be delivered to them from users of the
> connected server. And let them know you could add exceptions to be either
> blacklisted or whitelisted for blocked or allowed users respectively if
> necessary under settings.
> Here's a scenario. I am user A. I login to Team Talk on Server 1. User B, C
> and D are online in server 1. User B, I don't know. User C, I do. And user D
> is friends with user C.
> Assume I as user A hang out quite frequently on Server 1. But not all users
> connected to Server 1 know this. User D is such a user. But I don't know
> user D yet but Users B and C do.
> A block on all private messages means that User D can't get to know the only
> person they may not know on a particular friend's server, using Server 1 as
> an example. Allowing all private messages means that I can't block user D
> from getting to know me as I'm user A. Furthermore, Users B and C may get
> upset if I'm uninterested in what user D has to say, therefore causing
> needless head aches. Because blocking private messages will just mean the
> public chat in Server 1 would get used instead to draw attention. Yes you
> can disable messaging altogether in preferences, but introducing a block on
> all private messages by default will defeat the purpose of Team Talk in the
> first place. It is not a silent hang out zone. There needs to be mutual
> intelligibility.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/BearWare/TeamTalk5/discussions/1806#discussioncomment-5552430
> You are receiving this because you are subscribed to this thread.
>
> Message ID:
> ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili
